core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -834,6 +834,15 @@ public NativeImageInvokerInfo build() {
 
                 addExperimentalVMOption(nativeImageArgs, "-H:+AllowFoldMethods");
 
+                /*
+                 * Foreign Function and Memory API in Native Image, JDK's JEP 454
+                 * This is needed for JDK 24+ internal native calls due to AWT.
+                 * TODO: Do we enable it globally like so or do we only enable it when AWT extension is used?
```

Comment:
I think I would rather be consistent rather than enabling obscure options only when a very specific extension is enabled.

So +1 for your current approach but I also would like @zakkak 's opinion on this.

Reply:
```suggestion
```